### PR TITLE
Make drag n drop smoother

### PR DESCRIPTION
**Pull request by teldosas**
Following [this example](https://react-dnd.github.io/react-dnd/examples-sortable-simple.html) ([source](https://github.com/react-dnd/react-dnd/tree/master/examples/04%20Sortable/Simple)) I made the drag n drop smoother 
# [Check it out](https://teldosas.github.io/react-tags/example/)

### CODE REVIEW

**Comment by prakhar1989** [comment]
Hey @teldosas: This is awesome! I remember wanting to implement this, but at that time ReactDND didn't have this example. Can you please resolve the conflicts before I can review this? I'd suggest not adding any file from `dist-*` and `dist` folder in this PR.

**Comment by teldosas** [comment]
@prakhar1989 It should be ready now :)

File: lib/ReactTags.js [hunk 1]
```diff
@@ -233,27 +233,23 @@ class ReactTags extends Component {
...
     // locate tags
-    const tag = tags.filter(t => t.id === id)[0];
-    const afterTag = tags.filter(t => t.id === afterId)[0];
-
-    // find their position in the array
-    const tagIndex = tags.indexOf(tag);
-    const afterTagIndex = tags.indexOf(afterTag);
+    const dragTag = tags[dragIndex];
 
     // call handler with current position and after position
```
Comment (prakhar1989) [inline lib/ReactTags.js]: Change comment

File: lib/Tag.js [hunk 2]
```diff
@@ -1,21 +1,58 @@
...
+      // Get horizontal middle
+      const hoverMiddleX = (hoverBoundingRect.right - hoverBoundingRect.left) / 2;
+
+      // Determine mouse position
+      const clientOffset = monitor.getClientOffset();
+
+      // Get pixels to the left side
+      const hoverClientX = clientOffset.x - hoverBoundingRect.left;
+
+      // Only perform the move when the mouse has crossed half of the items width
```
Comment (prakhar1989) [inline lib/Tag.js]: I think all comments apart from this are fairly straightforward and can be removed. Plus, we also have the ReactDND example to refer to. What do you think?

**Comment by prakhar1989** [comment]
I wish the ReactDND examples included some example for testing these interactions. The current test suite lacks any unit tests for this behavior 😥

File: lib/Tag.js (hunk 2 above)
Comment (prakhar1989) [inline lib/Tag.js]: It doesn't hurt but adding comments for code that is clear by itself serves no purpose.

**Comment by teldosas** [comment]
It should be ready :)

**Comment by prakhar1989** [comment]
Great! Thanks for the addressing the comments!

**Pull request merged**